test/testother.cpp

```diff
@@ -5088,6 +5089,30 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void testMisusedScopeObjectNamespace() {
+        check("namespace M {\n" // #4479
+              "    namespace N {\n"
+              "        struct S {};\n"
+              "    }\n"
+              "}\n"
+              "int f() {\n"
+              "    M::N::S();\n"
+              "    return 0;\n"
```

Comment:
If the expression will be at the last line of function, it won't work (to reproduce: `"void f() {\n"` and remove `"return 0"`)

Reply:
Yes, that's some sort of heuristic apparently. As far as I'm concerned, it should be removed.